### DISCUSSION TITLE
Confusion about the api and query language

### DISCUSSION THREAD

**Discussion by hgftrdw45ud67is8o89** [orientechnologies/orientdb#10058] [Q&A] [answered]
Is gremlin just a api?
because from https://orientdb.org/docs/3.0.x/gremlin/Gremlin.html
It seems like its just a graphy* java api to query,am I correct?

While the general java api has graph api and a document api.

what is the difference?@@ I am confused.sorry.what api should I be using?

(for context)I wanted to use to create a database that support graph like relationship, if I am not wrong orientdb is not native graph but It kinda support query as a graph with a bit more overhead and syntax gimmicks.

EG:
I want to add 5 friend contact,and they have some info like name,birthday,age,location and relation

so I can search like,childhood friend of Adam and it returns list of name.

**Comment by andrii0lomakin** [comment, marked as answer]
Hi.

OrientDB is a native graph database.

Query language is also part of API, so there is no difference.

If you are asking what query language is better to use, those are general recommendations:
1. Only use Gremlin if you want API compatibility between different vendors. It is the least efficient from a performance point of view.
2. Prefer queries over other API methods if you work with (plan to migrate to) remote storage. Even though class-based API could be more handy, queries are typically executed with similar performance characteristics whether you work with embedded or remote storage.
